src/settings/target/mod.rs

```diff
@@ -266,88 +251,54 @@ impl Manifest {
         }
     }
 
-    // TODO: when --release is deprecated, this will be much easier
     fn negotiate_zoneless(
         &self,
         environment: Option<&Environment>,
-        release: bool,
     ) -> Result<(Option<String>, bool), failure::Error> {
-        let use_dot_dev_failure =
-            "Please specify the workers_dev boolean in the top level of your wrangler.toml.";
-        let use_dot_dev_warning =
-            format!("{}\n{} If you do not add workers_dev, this command may act unexpectedly in v1.5.0. Please see https://github.com/cloudflare/wrangler/blob/master/docs/content/environments.md for more information.", use_dot_dev_failure, emoji::WARN);
-        let wdd_failure = format!(
-            "{} Your environment should only include `workers_dev` or `route`. If you are trying to publish to workers.dev, remove `route` from your wrangler.toml, if you are trying to publish to your own domain, remove `workers_dev`.",
-            emoji::WARN
-        );
-
-        // TODO: deprecate --release, remove warnings and parsing
-        // switch wrangler publish behavior to act the same at top level
-        // and environments
-        // brace yourself, this is hairy
-        let workers_dev = match environment {
-            // top level configuration
-            None => {
-                if release {
-                    match self.workers_dev {
-                        Some(_) => {
-                            failure::bail!(format!("{} {}", emoji::WARN, use_dot_dev_failure))
-                        }
-                        None => {
-                            message::warn(&use_dot_dev_warning);
-                            false // wrangler publish --release w/o workers_dev is zoned deploy
-                        }
-                    }
-                } else if let Some(wdd) = self.workers_dev {
-                    if wdd {
-                        if let Some(route) = &self.route {
-                            if !route.is_empty() {
-                                failure::bail!(wdd_failure)
-                            }
+        let wdd_failure = "Your environment should only include `workers_dev` or `route`. If you are trying to publish to workers.dev, remove `route` from your wrangler.toml, if you are trying to publish to your own domain, remove `workers_dev`.";
+        let pick_target_failure =
+            "You must specify either `workers_dev` or `route` and `zone_id` in order to publish.";
+
+        if let Some(env) = &environment {
+            if let Some(wdd) = env.workers_dev {
```

Comment:
nit: the short hand keeps tripping me up.
```suggestion
            if let Some(workers_dev) = env.workers_dev {
```